`BUILD FAILED: Component #:SB-CGA not found, required by #<SYSTEM "nsb-cga">` issue while trying to use `sbclPackages.ncurses-clone-for-lem`

### Describe the bug
`sbclPackages.sb-cga` is probably missing, and hence my package build is failing with the `BUILD FAILED: Component #:SB-CGA not found, required by #<SYSTEM "nsb-cga">` issue. 

### Steps To Reproduce

1. Create a sample package that uses `ncurses-clone-for-lem`
```nix
{
  pkgs ? import <nixpkgs> {},
  stdenv ? pkgs.stdenv,
  fetchFromGitHub ? pkgs.fetchFromGitHub,
  sbcl' ? pkgs.sbcl.withPackages (ps: with ps; [ ncurses-clone-for-lem ]),
  tree ? pkgs.tree
}:
stdenv.mkDerivation rec {
  pname = "lem";
  version = "2.1.0";

  src = fetchFromGitHub {
    owner = "lem-project";
    repo = pname;
    rev = "v${version}";
    sha256 = "1bjm6z33n1ksjvqs5v1g1mz4f8bsx1lka85sspmlnsbswwap58k0";
  };

  buildInputs = [ sbcl' ];
  
  buildPhase = ''
    ${sbcl'}/bin/sbcl \
    --load scripts/patch-build-ncurses.lisp
  '';

  installPhase = ''
    echo "TODO: Add install instructions"
  '';
}
```
2. Try building the package with Nix

### Expected behavior
`ncurses-clone-for-lem` should use `sb-cga`, but for some reason, this dependency is missing.

### Screenshots
\-

### Additional context
I'm trying to build a package for [Lem](https://github.com/lem-project/lem). Here's the `nix log`:

```sh
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/x118r80mfv8rxk0nk479s240qn0h33b9-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
This is SBCL 2.3.4.nixos, an implementation of ANSI Common Lisp.
More information about SBCL is available at <http://www.sbcl.org/>.

SBCL is free software, provided as is, with absolutely no warranty.
It is mostly in the public domain; some portions are provided under
BSD-style licenses.  See the CREDITS and COPYING files in the
distribution for more information.
* #<PACKAGE "NIX-CL-USER">
* #<PACKAGE "NIX-CL-USER">
* T
* LOAD-SYSTEMS
* BUILD FAILED: Component #:SB-CGA not found, required by #<SYSTEM "nsb-cga">
```

### Notify maintainers
@Uthar
@7c6f434c
@lukego
@nagy
@hraban

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.43, NixOS, 23.05 (Stoat), 23.05.2561.9607b9149c9d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(ashvith): `""`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/development/lisp-modules/packages.nix
+++ b/pkgs/development/lisp-modules/packages.nix
@@ -799,6 +799,22 @@ let
     ];
   };
 
+  sb-cga = build-asdf-system {
+    pname = "sb-cga";
+    version = "1.0.1";
+    src = pkgs.fetchFromGitHub {
+      owner = "nikodemus";
+      repo = "sb-cga";
+      rev = "9a554ea1c01cac998ff7eaa5f767bc5bcdc4c094";
+      sha256 = "sha256-iBM+VXu6JRqGmeIFzfXbGot+elvangmfSpDB7DjFpPg";
+    };
+    lispLibs = [ super.alexandria ];
+  };
+
+  nsb-cga = super.nsb-cga.overrideLispAttrs (oa: {
+    lispLibs = oa.lispLibs ++ [ self.sb-cga ];
+  });
+
   });
 
 in packages

